service/src/test/java/bio/terra/user/app/controller/ProfileApiControllerTest.java

```diff
@@ -0,0 +1,111 @@
+package bio.terra.user.app.controller;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import bio.terra.common.iam.SamUser;
+import bio.terra.user.testutils.BaseUnitTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+class ProfileApiControllerTest extends BaseUnitTest {
+  private static final String API = "/api/profile";
+
+  @Autowired private MockMvc mockMvc;
+
+  @MockBean private ProfileSamUserFactory userFactory;
+
+  private final SamUser user = mock(SamUser.class);
+
+  @BeforeEach
+  void beforeEach() {
+    when(userFactory.from(any())).thenReturn(user);
+    var id = String.valueOf(Math.random()).substring(2);
+    when(user.getSubjectId()).thenReturn("fake-" + id);
+  }
+
+  @Test
+  void getEmptyProfile() throws Exception {
+    mockMvc.perform(get(API)).andExpect(status().isNotFound());
+  }
+
+  @Test
+  void recursiveSet() throws Exception {
+    String json = "{ \"value\": \"John\" }";
```

Comment:
It makes these tests easier to read if you write helper methods for the mockMvc calls ([example](https://cs.github.com/DataBiosphere/terra-workspace-manager/blob/792780ab6d1bf38ee9e4b811811952135974c7f2/service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java#L165))

```
  @Test
  void recursiveSet() throws Exception {
    setUserProfile("user.name.first", "John");
    assertUserProfile("user.name.first", "John");
  }
```

Also, consider combining tests if there's overlapping setup.

```
  @Test
  void setUserProfile() throws Exception {
    setUserProfile("user.name.first", "John");
    assertUserProfile("user.name.first", "John");

    // Existing property is updated
    setUserProfile("user.name.first", "Mary");
    assertUserProfile("user.name.first", "Mary");
  }
```